src/packages/engine-core/src/undici.ts

```diff
@@ -1,33 +1,36 @@
 import getStream = require('get-stream')
-import { Client, Pool } from 'undici'
+import { Client, Dispatcher, Pool } from 'undici'
 import { URL } from 'url'
 export class Undici {
   private pool: Pool
   private closed = false
+  private url: string | URL
```

Comment:
not sure about this one. `unidici` requires `origin` in `Dispatcher.DispatchOptions` even if it isn't required in their tests. I can remove this code and use `Omit<Dispatcher.DispatchOptions, 'origin'>` to fix that and get rid of this code.